File: mifosng-android/src/main/java/com/mifos/mifosxdroid/adapters/ClientReportAdapter.java
package com.mifos.mifosxdroid.adapters;

import androidx.recyclerview.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.mifos.mifosxdroid.R;
import com.mifos.objects.runreports.client.ClientReportTypeItem;

import java.util.List;

import javax.inject.Inject;

import butterknife.BindView;
import butterknife.ButterKnife;

/**
 * Created by Tarun on 03-08-17.
 */

public class ClientReportAdapter extends
        RecyclerView.Adapter<ClientReportAdapter.ViewHolder> {

    private List<ClientReportTypeItem> items;

    @Inject
    public ClientReportAdapter() {}

    public void setReportItems(List<ClientReportTypeItem> items) {
        this.items = items;
    }

    @Override
    public ClientReportAdapter.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_client_runreport, parent, false);

        return new ClientReportAdapter.ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(ClientReportAdapter.ViewHolder holder, int position) {
        if (holder != null) {
            holder.tvReportCategory.setText(items.get(position).getReportCategory());
            holder.tvReportName.setText(items.get(position).getReportName());
            holder.tvReportType.setText(items.get(position).getReportType());
        }
    }

    @Override
    public int getItemCount() {
        return items.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder {

        public ViewHolder(View v) {
            super(v);
            ButterKnife.bind(this, v);
        }

        @BindView(R.id.tv_report_name)
        TextView tvReportName;

        @BindView(R.id.tv_report_type)
        TextView tvReportType;

        @BindView(R.id.tv_report_category)
        TextView tvReportCategory;
    }

}

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/adapters/ClientReportAdapter.kt
package com.mifos.mifosxdroid.adapters

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.mifos.core.objects.runreports.client.ClientReportTypeItem
import com.mifos.mifosxdroid.R


class ClientReportAdapter(
    val onClientReportClick: (Int) -> Unit
) : RecyclerView.Adapter<ClientReportAdapter.ViewHolder>() {
    private var items: List<ClientReportTypeItem> = emptyList()


    fun setReportItems(items: List<ClientReportTypeItem>) {
        this.items = items
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val viewHolder = ViewHolder(
            LayoutInflater.from(parent.context)
                .inflate(R.layout.item_client_runreport, parent, false)
        )
        viewHolder.itemView.setOnClickListener {
            if (viewHolder.adapterPosition != RecyclerView.NO_POSITION)
                onClientReportClick(viewHolder.adapterPosition)
        }
        return viewHolder
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.tvReportCategory.text = items[position].report_category
        holder.tvReportName.text = items[position].report_name
        holder.tvReportType.text = items[position].report_type
    }

    override fun getItemCount() = items.size


    class ViewHolder(v: View) : RecyclerView.ViewHolder(v) {
        val tvReportName: TextView = v.findViewById(R.id.tv_report_name)
        val tvReportType: TextView = v.findViewById(R.id.tv_report_type)
        val tvReportCategory: TextView = v.findViewById(R.id.tv_report_category)
    }
}
